src/dlr_tvm.cc

```diff
@@ -214,3 +223,32 @@ void TVMModel::UseCPUAffinity(bool use) {
     LOG(INFO) << "CPU Affinity is disabled";
   }
 }
+
+bool TVMModel::HasMetadata() const {
+  return this->metadata != nullptr;
+}
+
+const char* TVMModel::GetOutputName(const int index) const {
+  std::string name_str = this->metadata["Model"]["Outputs"][index]["name"];
+  return name_str.c_str();
+}
+
+int TVMModel::GetOutputIndex(const char* name) const {
+  int idx = -1;
+
+  for (int i = 0; i < this->num_outputs_; i++) {
+    std::string name_str = this->metadata["Model"]["Outputs"][i]["name"];
+    if (strcmp(name, name_str.c_str()) == 0) {
+      idx = i;
+      break;
```

Comment:
just  `return i`

Reply:
This will cause issues i weird cases where the output names won't match or number of outputs come out to be zero. Also if uses passes bad output name it will always return the last index which is incorrect.

Reply:
We can do
```
for (...) {
  if (...) {
    return i;
  }
}
return -1;
```

Reply:
Same thing differently. Ok I will do it if it makes you happy ;)